subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/JvmLanguageUtilities.java

```diff
@@ -36,4 +38,13 @@ public interface JvmLanguageUtilities {
      * @param compileTask the compile task which serves as reference for inference
      */
     <COMPILE extends AbstractCompile & HasCompileOptions> void useDefaultTargetPlatformInference(Configuration configuration, TaskProvider<COMPILE> compileTask);
+
+    /**
+     * Registers a new source directory for a source set, assuming that it will be compiled by
```

Comment:
Thoughts on adding some sort of disclaimer to this API, perhaps even an `@Deprecated`? 

We should be sure that users who _do_ use this are aware it is in fact internal and subject to change/removal/replacement

Reply:
I didn't do this in order to hold to the "don't deprecate without a replacement available" guidance.  I think we deprecate when we address the linked issue about making a public API.

Reply:
It's in an `internal` package - that should be enough.  We've avoided deprecated other internal APIs in similar situations.

Reply:
Ususally I'd say thats enough as well. Though the fact that this code is referenced in [Understanding Gradle](https://github.com/jjohannes/understanding-gradle) could confuse people. 

I'm fine leaving this as-is